ui/src/app/components/copy-modal.tsx

```diff
@@ -4,22 +4,25 @@ import { Button } from 'app/components/buttons';
 import { styles as headerStyles } from 'app/components/headers';
 import { Select, TextInput, ValidationError } from 'app/components/inputs';
 import { Modal, ModalBody, ModalFooter, ModalTitle } from 'app/components/modals';
-import { FileDetail, Workspace } from 'generated/fetch';
+import { Workspace } from 'generated/fetch';
 
 import { Spinner } from 'app/components/spinners';
-import {appendNotebookFileSuffix, dropNotebookFileSuffix} from 'app/pages/analysis/util';
 import { workspacesApi } from 'app/services/swagger-fetch-clients';
 import { navigate } from 'app/utils/navigation';
+import {ResourceType, ResourceTypeDisplayNames} from 'app/utils/resourceActionsReact';
 import { WorkspacePermissions } from 'app/utils/workspace-permissions';
 
 enum RequestState { UNSENT, ERROR, SUCCESS }
 
 export interface Props {
+  destinationTab: string;
   fromWorkspaceNamespace: string;
   fromWorkspaceName: string;
-  fromNotebook: FileDetail;
+  fromResourceName: string;
   onClose: Function;
   onCopy: Function;
+  resourceType: ResourceType;
+  saveFunction: (CopyRequest) => Promise<any>;
```

Comment:
can we type the `any`?

Reply:
also maybe `copy` or `copyFunction` might be a more appropriate name? I think I prefer dropping the `Function` suffix if we're good about applying types.

Reply:
We can't, because it could return any number of objects. This is explicitly intended to be generic. 

Happy to drop the suffix and name it copy though.

Reply:
I guess I'm confused as to how we've agreed on the shape of a `CopyRequest` but still need to leave the return type unspecified. Could there not be a `CopyResponse` that has an array of the copies created, or whatever? Just thinking out loud.

Reply:
I actually figured out how to do the typing, so we're actually good on that! :D